src/utils/db/sql.ts

```diff
@@ -51,6 +52,25 @@ export async function getPluginConfigRows(server: PluginsServer): Promise<Plugin
     return rows
 }
 
+export async function setPluginCapabilities(
+    server: PluginsServer,
+    pluginConfig: PluginConfig,
+    capabilities: Capabilities
+): Promise<void> {
+    await server.db.postgresQuery(
+        'UPDATE posthog_plugin SET capabilities = ($1) WHERE id = $2',
+        [capabilities, pluginConfig.plugin_id],
+        'setPluginCapabilities'
+    )
+    await server.db.createPluginLogEntry(
+        pluginConfig,
+        PluginLogEntrySource.System,
+        PluginLogEntryType.Info,
+        `Set plugin capabilities (instance ID ${server.instanceId}).`,
+        server.instanceId
+    )
```

Comment:
I think if the capabilities haven't changed, there's no need to update them nor log that we set them. That would be needless logspam :)